app/controllers/tags_controller.rb

```diff
@@ -4,11 +4,23 @@ class TagsController < ApplicationController
   after_action :verify_authorized
 
   ATTRIBUTES_FOR_SERIALIZATION = %i[id name bg_color_hex text_color_hex].freeze
+  INDEX_API_ATTRIBUTES = %i[name rules_html].freeze
 
   def index
     skip_authorization
     @tags_index = true
-    @tags = Tag.includes(:sponsorship).order(hotness_score: :desc).limit(100)
+    tags = Tag.order(hotness_score: :desc).limit(100)
+
+    respond_to do |format|
```

Comment:
@citizen428 I have a general-ish question regarding this, it feels unusual that we return different info to the html format format vs the json format. I would have expected that the data that we return would be the same and only the format to differ, but maybe this is just the nature of the apps I've worked on. 

Is it a common practice to return different json vs html?

Reply:
> I have a general-ish question regarding this, it feels unusual that we return different info to the html format format vs the json format.

That's a fair observation. In a way, we do return the same data, as we start from the same AR relation. Then for each format, we provide all the information that is necessary to display the information, which in the case of HTML is a bit more for the serverside rendering, whereas in the case of JSON it's just two attributes. Sending all attributes over the wire if we don't need them seemed a bit wasteful.

Reply:
We talked through this on Slack and in the end I decided to move it to a custom route.